<a href="https://colab.research.google.com/github/aerab900/datasci_1_loading-/blob/main/HHA507_Assignment%231.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**The following code cell below imports pandas**  

We need pandas because we cannot load our datasets without it  


In [5]:
import pandas as pd


## Dataset 1: Online Food Delivery (CSV)

This dataset is survey data from customers who use online food delivery services in Bangalore, India.

**Features (columns):**

- **Age**: the customer's age in years (roughly 18–33)
- **Gender**: Male or Female
- **Marital Status**: Single, Married, or Prefer not to say
- **Occupation**: the customer's job type (Student, Employee, Self Employed, or House wife)
- **Monthly Income**: income range in Indian rupees (e.g., No Income, Below Rs.10000, 10001 to 25000, More than 50000)
- **Educational Qualifications**: highest education level (School, Graduate, Post Graduate, Ph.D, or Uneducated)
- **Family size**: number of people in the customer's household (1–6)
- **Customer Type**: how often the customer orders (New, Regular, or Frequent)
- **latitude**: location coordinate of the customer
- **longitude**: location coordinate of the customer
- **Pin code**: the customer's zip code
- **Output**: Yes/No answer to whether the customer would order again
- **Feedback**: whether the customer's review was Positive or Negative
- **Unnamed: 13**: this column has no header in the file and looks like a duplicate of the Output column (Yes/No)


In [36]:
df1 = '/content/Data/online food delivery dataset.csv'
read_df1 = pd.read_csv(df1)
read_df1.head(11)


,Age,Gender,Marital Status,Occupation,Monthly Income,Educational Qualifications,Family size,Customer Type,latitude,longitude,Pin code,Output,Feedback,Unnamed: 13
0,20,Female,Single,Student,No Income,Post Graduate,4,Frequent,12.9766,77.5993,560001,Yes,Positive,Yes
1,24,Female,Single,Student,Below Rs.10000,Graduate,3,Regular,12.9770,77.5773,560009,Yes,Positive,Yes
2,22,Male,Single,Student,Below Rs.10000,Post Graduate,3,Regular,12.9551,77.6593,560017,Yes,Negative,Yes
3,22,Female,Single,Student,No Income,Graduate,6,Frequent,12.9473,77.5616,560019,Yes,Positive,Yes
4,22,Male,Single,Student,Below Rs.10000,Post Graduate,4,Frequent,12.9850,77.5533,560010,Yes,Positive,Yes
5,27,Female,Married,Employee,More than 50000,Post Graduate,2,Regular,12.9299,77.6848,560103,Yes,Positive,Yes
6,22,Male,Single,Student,No Income,Graduate,3,Regular,12.9770,77.5773,560009,Yes,Positive,Yes
7,24,Female,Single,Student,No Income,Post Graduate,3,Regular,12.9828,77.6131,560042,Yes,Positive,Yes
8,23,Female,Single,Student,No Income,Post Graduate,2,Regular,12.9766,77.5993,560001,Yes,Positive,Yes
9,23,Female,Single,Student,No Income,Post Graduate,4,Frequent,12.9854,77.7081,560048,Yes,Positive,Yes


## Dataset 2: AI Impact on Jobs & Salaries (CSV)

This dataset contains self-reported salary survey data for AI, machine learning, and data jobs. Each row is one reported salary and there are approximately 72,000 rows.

**Features (columns):**

- **work_year**: the year the salary was reported
- **experience_level**: short code for experience level (EN = Entry, MI = Mid, SE = Senior, EX = Executive)
- **employment_type**: short code for employment type (FT = Full-time, PT = Part-time, CT = Contract, FL = Freelance)
- **job_title**: the job title as the person reported it
- **salary**: the salary amount in its original currency
- **salary_currency**: the currency the salary was paid in (e.g., USD, EUR)
- **salary_in_usd**: the salary converted to U.S. dollars so all rows can be compared
- **employee_residence**: the country where the employee lives
- **remote_ratio**: how much of the job is remote (0 = on-site, 50 = hybrid, 100 = fully remote)
- **company_location**: the country where the company is located
- **company_size**: size of the company (S = small, M = medium, L = large)
- **experience_level_label**: the experience level code written out in words
- **employment_type_label**: the employment type code written out in words (e.g., Full-time)
- **work_mode**: remote_ratio writtein in word format (Remote, Hybrid, Onsite)
- **salary_outlier_flag**: True if the salary is unusually high or low compared to the rest of the data
- **role_family**: a broader job category grouped from job_title
- **isco_group_hint**: an international occupation category

In [9]:
df2 = '/content/Data/ai_jobs_salaries.csv'
read_df2 = pd.read_csv(df2)
read_df2.head(11)


,work_year,experience_level,employment_type,job_title,salary,salary_currency,salary_in_usd,employee_residence,remote_ratio,company_location,company_size,experience_level_label,employment_type_label,work_mode,salary_outlier_flag,role_family,isco_group_hint
0,2025,EX,FT,Head of Data,348516,USD,348516,US,0,US,M,Executive-level,Full-time,On-site,True,Analytics Manager,Information and communications technology serv...
1,2025,EX,FT,Head of Data,232344,USD,232344,US,0,US,M,Executive-level,Full-time,On-site,False,Analytics Manager,Information and communications technology serv...
2,2025,SE,FT,Data Scientist,145400,USD,145400,US,0,US,M,Senior-level,Full-time,On-site,False,Data Scientist,"Mathematicians, actuaries and statisticians"
3,2025,SE,FT,Data Scientist,81600,USD,81600,US,0,US,M,Senior-level,Full-time,On-site,False,Data Scientist,"Mathematicians, actuaries and statisticians"
4,2025,MI,FT,Engineer,160000,USD,160000,US,100,US,M,Mid-level,Full-time,Remote,False,Other / Unclassified,NaN
5,2025,MI,FT,Engineer,140000,USD,140000,US,100,US,M,Mid-level,Full-time,Remote,False,Other / Unclassified,NaN
6,2025,SE,FT,AI Product Lead,175000,USD,175000,US,100,US,M,Senior-level,Full-time,Remote,False,Other / Unclassified,NaN
7,2025,SE,FT,AI Product Lead,152900,USD,152900,US,100,US,M,Senior-level,Full-time,Remote,False,Other / Unclassified,NaN
8,2025,SE,FT,AI Engineer,97900,USD,97900,US,100,US,M,Senior-level,Full-time,Remote,False,AI Engineer,Software and applications developers
9,2025,SE,FT,AI Engineer,89900,USD,89900,US,100,US,M,Senior-level,Full-time,Remote,False,AI Engineer,Software and applications developers


## Dataset 3: College Major ROI (CSV)

This dataset looks at the return on investment (ROI) of a college degree. It compares what graduates earn over 10 years. Each row is a college graduate.

**Features (columns):**

- **grad_id**: a unique ID number for each graduate
- **major**: the graduate's specific major (19 majors listed)
- **major_category**: the broader group the major belongs to (STEM, health, business, social_science, education, humanities, arts)
- **institution_tier**: the type of school attended (ivy_elite, top_public, mid_tier, regional, for_profit)
- **institution_selectivity_pctile**: how selective the school is in admissions, as a percantage (higher percentage = harder to get in)
- **gpa**: the graduate's grade point average on a 0–4 scale
- **had_internship**: 1 = yes. 0 = no.
- **completed_on_time**: 1 = yes, completed on time. 0 = no, not completed on time
- **region**: the U.S. region (northeast, south, midwest, west)
- **net_cost_usd**: the total cost of the degree after financial aid, in U.S. dollars
- **debt_usd**: the amount of student loan debt taken on, in U.S. dollars (0 if none)
- **hs_baseline_10yr_usd**: how much the person would have earned over 10 years with only a high school diploma
- **added_earnings_10yr_usd**: the extra money earned over 10 years because of the degree (the degree's "earnings premium")
- **earnings_10yr_usd**: total earnings over 10 years after graduating, in U.S. dollars
- **net_roi_usd**: the degree's net return in dollars (added earnings minus net cost)
- **roi_pct**: the net ROI as a percent of the net cost
- **positive_roi**: 1 if the degree paid off (net_roi_usd is greater than 0), 0 if not
- **high_roi**: 1 if the graduate's net ROI is in the top 25% of the dataset, 0 if not

In [12]:
df3 = '/content/Data/college_major_roi.csv'
read_df3 = pd.read_csv(df3)
read_df3.head(11)


,grad_id,major,major_category,institution_tier,institution_selectivity_pctile,gpa,had_internship,completed_on_time,region,net_cost_usd,debt_usd,hs_baseline_10yr_usd,added_earnings_10yr_usd,earnings_10yr_usd,net_roi_usd,roi_pct,positive_roi,high_roi
0,G000000,biology,STEM,mid_tier,52,2.58,0,0,south,144400,0,411700,101000,512700,-43400,-30.0,0,0
1,G000001,marketing,business,mid_tier,61,2.58,0,1,northeast,57400,0,356900,330600,687500,273200,476.1,1,0
2,G000002,electrical_engineering,STEM,top_public,78,3.14,1,1,northeast,65100,39600,339600,655100,994700,590000,906.4,1,1
3,G000003,marketing,business,top_public,77,3.04,0,1,midwest,105800,81300,399600,197500,597100,91700,86.8,1,0
4,G000004,marketing,business,ivy_elite,100,3.81,0,0,south,219200,93500,335700,222900,558600,3800,1.7,1,0
5,G000005,business,business,regional,37,3.91,1,0,northeast,79000,0,413800,466800,880600,387900,491.2,1,0
6,G000006,economics,business,regional,44,2.32,1,1,midwest,39500,0,388800,405200,794100,365800,926.3,1,0
7,G000007,social_work,social_science,mid_tier,59,4.00,1,1,midwest,59300,43500,350300,162400,512600,103100,174.0,1,0
8,G000008,mechanical_engineering,STEM,regional,45,3.61,0,1,northeast,62200,0,314300,344500,658800,282300,454.2,1,0
9,G000009,finance,business,top_public,79,2.46,1,0,south,109000,107500,307200,438000,745200,329000,301.8,1,0


## Dataset 4: Financial Sample (XLSX)

This dataset contains made-up sales and profit data for a company that sells six products in five countries. It is Microsoft's sample file for learning Power BI. There are 700 rows, and each row is one sale record.

**Features (columns):**

- **Segment**: the type of customer (Government, Midmarket, Channel Partners, Enterprise, Small Business)
- **Country**: where the sale happened (Canada, France, Germany, Mexico, United States of America)
- **Product**: the product sold (Carretera, Montana, Paseo, Velo, VTT, Amarilla)
- **Discount Band**: the level of discount given (None, Low, Medium, High)
- **Units Sold**: how many units were sold
- **Manufacturing Price**: the cost to make one unit, in dollars
- **Sale Price**: the price one unit was sold for, in dollars
- **Gross Sales**: total sales before discounts (Units Sold × Sale Price)
- **Discounts**: the dollar amount taken off for discounts
- **Sales**: total sales after discounts (Gross Sales minus Discounts)
- **COGS**: Cost of Goods Sold, the total cost to make the units that were sold
- **Profit**: money made after costs (Sales minus COGS)
- **Date**: the date of the sale
- **Month Number**: the month as a number (1–12)
- **Month Name**: the month written out (e.g., January)
- **Year**: the year of the sale (2013 or 2014)

In [13]:
df4 = '/content/Data/Financial Sample.xlsx'
read_df4 = pd.read_excel(df4)
read_df4.head(11)


,Segment,Country,Product,Discount Band,Units Sold,Manufacturing Price,Sale Price,Gross Sales,Discounts,Sales,COGS,Profit,Date,Month Number,Month Name,Year
0,Government,Canada,Carretera,NaN,1618.5,3,20,32370.0,0.0,32370.0,16185.0,16185.0,2014-01-01,1,January,2014
1,Government,Germany,Carretera,NaN,1321.0,3,20,26420.0,0.0,26420.0,13210.0,13210.0,2014-01-01,1,January,2014
2,Midmarket,France,Carretera,NaN,2178.0,3,15,32670.0,0.0,32670.0,21780.0,10890.0,2014-06-01,6,June,2014
3,Midmarket,Germany,Carretera,NaN,888.0,3,15,13320.0,0.0,13320.0,8880.0,4440.0,2014-06-01,6,June,2014
4,Midmarket,Mexico,Carretera,NaN,2470.0,3,15,37050.0,0.0,37050.0,24700.0,12350.0,2014-06-01,6,June,2014
5,Government,Germany,Carretera,NaN,1513.0,3,350,529550.0,0.0,529550.0,393380.0,136170.0,2014-12-01,12,December,2014
6,Midmarket,Germany,Montana,NaN,921.0,5,15,13815.0,0.0,13815.0,9210.0,4605.0,2014-03-01,3,March,2014
7,Channel Partners,Canada,Montana,NaN,2518.0,5,12,30216.0,0.0,30216.0,7554.0,22662.0,2014-06-01,6,June,2014
8,Government,France,Montana,NaN,1899.0,5,20,37980.0,0.0,37980.0,18990.0,18990.0,2014-06-01,6,June,2014
9,Channel Partners,Germany,Montana,NaN,1545.0,5,12,18540.0,0.0,18540.0,4635.0,13905.0,2014-06-01,6,June,2014


## Dataset 5: Retail Inventory (XLSX)

This dataset contains made-up inventory records for a retail store. There are 1,000 rows, and each row is one product.

**Features (columns):**

- **Product ID**: a unique code for each product (e.g., PRD-1001)
- **Product Name**: the name of the product
- **Category**: the type of product (e.g., Home & Kitchen, Beauty, Electronics, Grocery)
- **Supplier Name**: the company that supplies the product
- **Unit Cost**: what the store pays for one unit, in dollars
- **Retail Price**: what the store sells one unit for, in dollars
- **Opening Stock**: how many units were in stock at the start of the period
- **Units Sold**: how many units were sold during the period
- **Closing Stock**: how many units were left at the end of the period (Opening Stock minus Units Sold)
- **Reorder Point**: the stock level at which the store should order more
- **Stock Status**: whether the product is In Stock or Low Stock (Low Stock means Closing Stock is at or below the Reorder Point)

In [14]:
df5 = '/content/Data/Retail Inventory.xlsx'
read_df5 = pd.read_excel(df5)
read_df5.head(11)


,Product ID,Product Name,Category,Supplier Name,Unit Cost,Retail Price,Opening Stock,Units Sold,Closing Stock,Reorder Point,Stock Status
0,PRD-1001,Wine Glasses,Home & Kitchen,Heritage Brands,23.78,38.89,135,118,17,30,Low Stock
1,PRD-1002,Eye Mask,Beauty,Coastal Trading,15.67,24.53,86,10,76,9,In Stock
2,PRD-1003,Resistance Band,Sports & Outdoors,National Supply Group,5.59,11.88,319,195,124,47,In Stock
3,PRD-1004,Action Figure,Toys & Games,Pacific Wholesale,9.72,18.01,173,140,33,39,Low Stock
4,PRD-1005,Serving Tray,Home & Kitchen,Coastal Trading,8.22,13.90,32,8,24,7,In Stock
5,PRD-1006,Dried Fruit,Grocery,National Supply Group,6.94,12.18,74,52,22,15,In Stock
6,PRD-1007,Dish Soap,Home & Kitchen,Pacific Wholesale,6.74,13.47,21,6,15,2,In Stock
7,PRD-1008,Hiking Socks,Sports & Outdoors,National Supply Group,7.23,9.67,353,72,281,51,In Stock
8,PRD-1009,Dried Fruit,Grocery,Summit Wholesale Inc,10.60,16.76,196,59,137,26,In Stock
9,PRD-1010,Yoga Block,Sports & Outdoors,Premier Merchandise,5.54,9.54,38,14,24,4,In Stock


## Dataset 6: Project Management (XLSX)

This dataset contains made-up project tracking data for a company's internal projects. There are 1,000 rows, and each row is one task within a project.

**Features (columns):**

- **Project Name**: the project the task belongs to (30 projects, e.g., Security Audit, Website Redesign, Customer Portal)
- **Task Name**: the specific task being worked on (e.g., Bug Fixes, Risk Assessment, API Development)
- **Assigned To**: the team member responsible for the task
- **Priority**: how important the task is (Low, Medium, High, Critical)
- **Planned Start Date**: the date the task was scheduled to begin
- **Planned End Date**: the date the task was scheduled to be finished
- **Actual End Date**: the date the task was actually finished (blank if the task isn't done yet)
- **% Complete**: how much of the task is finished, shown as a decimal (0 = not started, 1 = 100% done)
- **Planned Budget**: the amount of money set aside for the task, in dollars
- **Actual Cost**: the amount of money actually spent on the task so far, in dollars
- **Status**: where the task stands (Not Started, In Progress, Completed, Delayed, On Hold, Cancelled)

In [20]:
df6 = '/content/Data/02 Project Management.xlsx'
read_df6 = pd.read_excel(df6)
read_df6.head(11)

,Project Name,Task Name,Assigned To,Priority,Planned Start Date,Planned End Date,Actual End Date,% Complete,Planned Budget,Actual Cost,Status
0,Compliance Overhaul,Deployment Planning,Chris Evans,Medium,2024-04-05,2024-04-27,2024-05-24,1.00,40000,40800,Completed
1,Marketing Automation,Bug Fixes,Brian Taylor,Medium,2024-01-04,2024-01-24,NaT,0.23,34000,17900,Delayed
2,Security Audit,Competitive Analysis,Elena Rodriguez,Low,2024-10-14,2024-10-21,2024-10-24,1.00,2000,1800,Completed
3,Email Platform Migration,Budget Review,Rachel Thompson,Medium,2024-04-24,2024-05-09,2024-04-28,0.30,18000,2700,Cancelled
4,Security Audit,Competitive Analysis,Natasha Singh,Medium,2025-01-10,2025-02-12,2025-02-11,1.00,31000,32500,Completed
5,Inventory System,Risk Assessment,Hannah White,Critical,2025-01-25,2025-02-28,2025-02-25,1.00,100000,116000,Completed
6,Onboarding Automation,System Configuration,Emily Davis,High,2024-11-19,2024-12-05,2024-11-23,0.10,44000,12900,Cancelled
7,Budget Planning Tool,A/B Testing Setup,Jennifer Lee,Medium,2024-06-10,2024-06-19,2024-06-14,0.10,14000,1200,Cancelled
8,Inventory System,SEO Optimization,James Wilson,Low,2024-04-07,2024-05-04,2024-04-13,0.10,16000,4500,Cancelled
9,Analytics Dashboard,Frontend Development,Kevin O'Brien,Low,2024-06-26,2024-09-12,2024-09-23,1.00,149000,198400,Completed


## Dataset 7: Top Websites by Country (JSON)

This dataset lists the 50 most-visited websites in 118 countries. There are 50 rows and 118 columns. Each column is a country, and each row is a popularity rank, so row 0 is the #1 most-visited site in that country and row 49 is #50.

**Features (columns):**

- **Index (0–49)**: the website's rank in that country, starting at 0 for the most popular site
- **Country columns (118 total)**: each column is named after a country (e.g., Afghanistan, Brazil, India, United States, Zimbabwe) and lists that country's top 50 websites in order of popularity



In [30]:
df7 = '/content/Data/countries.json'
read_df7 = pd.read_json(df7)
read_df7.head(11)


,Afghanistan,Albania,Algeria,Argentina,Armenia,Australia,Austria,Azerbaijan,Bahamas,Bahrain,...,Ukraine,United Arab Emirates,United Kingdom,United States,Uruguay,Uzbekistan,Venezuela,Vietnam,Yemen,Zimbabwe
0,Google.com,Google.com,Google.com,Google.com,Google.com,Google.com,Google.com,Google.com,Google.com,Google.com,...,Google.com,Google.com,Google.com,Google.com,Google.com,Google.com,Google.com,Google.com,Google.com,Google.com
1,Youtube.com,Youtube.com,Youtube.com,Youtube.com,Youtube.com,Youtube.com,Youtube.com,Youtube.com,Youtube.com,Youtube.com,...,Youtube.com,Youtube.com,Youtube.com,Youtube.com,Youtube.com,Youtube.com,Youtube.com,Youtube.com,Youtube.com,Youtube.com
2,Facebook.com,Gjirafa.com,Ouedkniss.com,Google.com.ar,Bongacams.com,Facebook.com,Google.at,Wikipedia.org,Yahoo.com,Office.com,...,Google.com.ua,Facebook.com,Reddit.com,Amazon.com,Mercadolibre.com.uy,Olx.uz,Patria.org.ve,Facebook.com,Facebook.com,Zimeye.net
3,Wikipedia.org,Gazetaexpress.com,Facebook.com,Mercadolibre.com.ar,List.am,Google.com.au,Amazon.de,Google.az,Bing.com,Google.com.bh,...,Bongacams.com,Google.ae,Amazon.co.uk,Yahoo.com,Google.com.uy,Soliq.uz,Banvenez.com,Shopee.vn,Elyamnelaraby.com,Nehandaradio.com
4,Acbar.org,Panorama.com.al,Google.dz,Facebook.com,Ok.ru,Myshopify.com,Reddit.com,Microsoft.com,Amazon.com,Microsoftonline.com,...,Olx.ua,Amazon.ae,Google.co.uk,Zoom.us,Brou.com.uy,Mail.ru,Google.co.ve,Google.com.vn,Alayyam.info,Bongacams.com
5,Khaama.com,Syri.net,Yahoo.com,Live.com,Google.ru,Ebay.com.au,Wikipedia.org,Mail.ru,Facebook.com,Microsoft.com,...,Wikipedia.org,Manoramaonline.com,Bbc.co.uk,Facebook.com,Netflix.com,Wikipedia.org,Bancodevenezuela.com,Vnexpress.net,Blogger.com,Yahoo.com
6,Bbc.com,Balkanweb.com,Ennaharonline.com,Yahoo.com,Wikipedia.org,Reddit.com,Willhaben.at,Tap.az,Microsoft.com,Edunet.bh,...,Ukr.net,Yahoo.com,Wikipedia.org,Reddit.com,Facebook.com,Kun.uz,Banesconline.com,Laodong.vn,Yemen.net.ye,Netflix.com
7,Stackoverflow.com,Dailyxing.com,Google.fr,Netflix.com,Facebook.com,Live.com,Orf.at,Facebook.com,Zoom.us,Live.com,...,Rozetka.com.ua,Live.com,Ebay.co.uk,Wikipedia.org,Elpais.com.uy,Kundalik.com,Facebook.com,Thethao247.vn,Wikipedia.org,Vacancymail.co.zw
8,Yahoo.com,Facebook.com,Wikipedia.org,Afip.gob.ar,Mail.ru,Netflix.com,Facebook.com,Yahoo.com,Lotterypost.com,Facebook.com,...,Privatbank.ua,Zoom.us,Bongacams.com,Myshopify.com,Live.com,Ok.ru,Mercadolibre.com.ve,Vtv.vn,Sahafah24.net,Office.com
9,Aparat.com,Classlifestyle.com,Ency-education.com,Infobae.com,News.am,Yahoo.com,Bongacams.com,Turbo.az,Office.com,Bahrain.bh,...,Vk.com,Amazon.com,Live.com,Ebay.com,Amazon.com,Google.ru,Mercantilbanco.com,24h.com.vn,Mediafire.com,Researchgate.net


## Dataset 8: Cars (JSON)

This dataset contains information about car models sold from 1970 to 1982, including their fuel efficiency, engine size, and performance. There are 406 rows, and each row is one car model. Source: Vega datasets (GitHub), originally from the UCI Machine Learning Repository "Auto MPG" dataset.

**Features (columns):**

- **Name**: the make and model of the car (e.g., ford pinto, toyota corolla). "(sw)" means station wagon.
- **Miles_per_Gallon**: fuel efficiency, meaning how many miles the car can drive on one gallon of gas
- **Cylinders**: the number of cylinders in the engine (3, 4, 5, 6, or 8). More cylinders usually means a bigger, more powerful engine.
- **Displacement**: the total size of the engine, in cubic inches
- **Horsepower**: how powerful the engine is
- **Weight_in_lbs**: the weight of the car, in pounds
- **Acceleration**: the number of seconds it takes the car to go from 0 to 60 mph (a lower number means faster)
- **Year**: the model year of the car (stored as a date, e.g., 1970-01-01 means a 1970 model)
- **Origin**: where the car was made (USA, Europe, or Japan)

In [32]:
df8 = '/content/Data/cars.json'
read_df8 = pd.read_json(df8)
read_df8.head(11)


,Name,Miles_per_Gallon,Cylinders,Displacement,Horsepower,Weight_in_lbs,Acceleration,Year,Origin
0,chevrolet chevelle malibu,18.0,8,307.0,130.0,3504,12.0,1970-01-01,USA
1,buick skylark 320,15.0,8,350.0,165.0,3693,11.5,1970-01-01,USA
2,plymouth satellite,18.0,8,318.0,150.0,3436,11.0,1970-01-01,USA
3,amc rebel sst,16.0,8,304.0,150.0,3433,12.0,1970-01-01,USA
4,ford torino,17.0,8,302.0,140.0,3449,10.5,1970-01-01,USA
5,ford galaxie 500,15.0,8,429.0,198.0,4341,10.0,1970-01-01,USA
6,chevrolet impala,14.0,8,454.0,220.0,4354,9.0,1970-01-01,USA
7,plymouth fury iii,14.0,8,440.0,215.0,4312,8.5,1970-01-01,USA
8,pontiac catalina,14.0,8,455.0,225.0,4425,10.0,1970-01-01,USA
9,amc ambassador dpl,15.0,8,390.0,190.0,3850,8.5,1970-01-01,USA


## Dataset 9: Palmer Penguins (JSON)

This dataset contains body measurements of penguins studied on three islands near Palmer Station, Antarctica. There are 344 rows, and each row is one penguin.

**Features (columns):**

- **Species**: the type of penguin (Adelie, Chinstrap, or Gentoo)
- **Island**: the island where the penguin was studied (Torgersen, Biscoe, or Dream)
- **Beak Length (mm)**: how long the penguin's beak is, in millimeters
- **Beak Depth (mm)**: how thick (top to bottom) the penguin's beak is, in millimeters
- **Flipper Length (mm)**: how long the penguin's flipper (wing) is, in millimeters
- **Body Mass (g)**: how much the penguin weighs, in grams
- **Sex**: whether the penguin is MALE or FEMALE

In [35]:
df9 = '/content/Data/penguins.json'
read_df9 = pd.read_json(df9)
read_df9.head(11)


,Species,Island,Beak Length (mm),Beak Depth (mm),Flipper Length (mm),Body Mass (g),Sex
0,Adelie,Torgersen,39.1,18.7,181.0,3750.0,MALE
1,Adelie,Torgersen,39.5,17.4,186.0,3800.0,FEMALE
2,Adelie,Torgersen,40.3,18.0,195.0,3250.0,FEMALE
3,Adelie,Torgersen,NaN,NaN,NaN,NaN,None
4,Adelie,Torgersen,36.7,19.3,193.0,3450.0,FEMALE
5,Adelie,Torgersen,39.3,20.6,190.0,3650.0,MALE
6,Adelie,Torgersen,38.9,17.8,181.0,3625.0,FEMALE
7,Adelie,Torgersen,39.2,19.6,195.0,4675.0,MALE
8,Adelie,Torgersen,34.1,18.1,193.0,3475.0,None
9,Adelie,Torgersen,42.0,20.2,190.0,4250.0,None
